visualstudio-extension/AZSphereHardwareDefinitionTools/PortHardwareDefinitionHandler.cs

```
@@ -0,0 +1,94 @@
+      if (!isValidHwDef) {
+        await CreateAndDisplayInfoBarAsync("The current file is not a valid Hardware Definition. Open a valid Hardware Definition to port from", currentFilePath);
+        return;
+      }
+      var odmHwDefinitions = await HardwareDefinitionLanguageClient.Instance.GetAvailableOdmHardwareDefinitionsAsync(currentFileUri);
+      var actions = odmHwDefinitions.Select(odmHwDef => new InfoBarHyperlink(odmHwDef.Name, new OdmHardwareDefinitionActionContext(currentFilePath, odmHwDef.Path))).ToList();
+      actions.Insert(0, new InfoBarHyperlink("Open new", new OdmHardwareDefinitionActionContext(currentFilePath, null)));
+      if (actions.Count > 0)
+      {
+        await CreateAndDisplayInfoBarAsync("Select a hardware definition to port to", currentFilePath, actions, OnOdmHardwareDefSelected);
+      }
+    }
+
+    private void OnOdmHardwareDefSelected(object sender, InfoBarActionItemEventArgs e)
+    {
+      ThreadHelper.ThrowIfNotOnUIThread();
+      CloseCurrentInfoBar();
+
+      OdmHardwareDefinitionActionContext selected = e.ActionItem.ActionContext;
+      if (selected.TargetHwDefPath == null)
+      {
+        using (OpenFileDialog openFileDialog = new OpenFileDialog())
+        {
+          openFileDialog.Filter = "json files (*.json)|*.json";
+          openFileDialog.RestoreDirectory = true;
+
+          if (openFileDialog.ShowDialog() == DialogResult.OK)
+          {
+            selected = new OdmHardwareDefinitionActionContext(selected.OriginalHwDefPath, openFileDialog.FileName);
+          }
+        }
+      }
+      _ = System.Threading.Tasks.Task.Run(async () =>
+      {
+        string originalHwDefFileName = Path.GetFileName(selected.OriginalHwDefPath);
+        string portedPath = await HardwareDefinitionLanguageClient.Instance.PortHardwareDefinitionAsync(selected.OriginalHwDefPath, selected.TargetHwDefPath);
```

Check TargetHwDef with ValidateHwDefinitionAsync, to inform the user why porting failed.
Or... in our current setup, porting can only fail if the TargetHwDef is invalid. (Am I not mistaken?)
If so, we can just change the Info message to tell the user that he is trying to port to an invalid HW Def.

---

We run the validate command on the source hw def beforehand so this should only fail if the target is invalid. Will update the info message